server/data/mappers/timelineMapper.ts

```diff
@@ -8,31 +8,39 @@ import type { Timeline, TimelineEvent } from 'viewModels'
  * As part of this mapping the timeline events are filtered to just those that are supported by the PLP UI
  * (The [TimelineResponse] API data includes event types that are not supported/required by the PLP UI)
  */
-const toTimeline = (timelineResponse: TimelineResponse): Timeline => {
+const toTimeline = (timelineResponse: TimelineResponse, prisonNamesById: Map<string, string>): Timeline => {
   return {
     problemRetrievingData: false,
     reference: timelineResponse.reference,
     prisonNumber: timelineResponse.prisonNumber,
-    events: timelineResponse.events.map(toTimelineEvent),
+    events: timelineResponse.events.map((event: TimelineEventResponse) => toTimelineEvent(event, prisonNamesById)),
   }
 }
 
-const toTimelineEvent = (timelineEventResponse: TimelineEventResponse): TimelineEvent => {
+const toTimelineEvent = (
+  timelineEventResponse: TimelineEventResponse,
+  prisonNamesById: Map<string, string>,
+): TimelineEvent => {
   return {
     reference: timelineEventResponse.reference,
     sourceReference: timelineEventResponse.sourceReference,
     eventType: timelineEventResponse.eventType,
-    prison: {
-      prisonId: timelineEventResponse.prisonId,
-      prisonName: undefined, // This mapper cannot look up / resolve the prisonName; that has to be post-processed
-    },
+    prisonName: prisonNamesById.get(timelineEventResponse.prisonId) || timelineEventResponse.prisonId,
     timestamp: toDate(timelineEventResponse.timestamp),
     correlationId: timelineEventResponse.correlationId,
-    contextualInfo: timelineEventResponse.contextualInfo,
+    contextualInfo: isPrisonTransfer(timelineEventResponse)
+      ? {
+          PRISON_TRANSFERRED_FROM:
+            prisonNamesById.get(timelineEventResponse.contextualInfo.PRISON_TRANSFERRED_FROM) ||
+            timelineEventResponse.contextualInfo.PRISON_TRANSFERRED_FROM,
+        }
+      : timelineEventResponse.contextualInfo,
```

Comment:
This feels like a bug in waiting if the prison transfer event ever has more contextual info. I kept what was being done originally but open to suggestions

Reply:
Yeah, thats a fair point 🤔 
I wonder if this would be better / cover that eventuality:
```
    contextualInfo: isPrisonTransfer(timelineEventResponse)
      ? {
          ...timelineEventResponse.contextualInfo,
          PRISON_TRANSFERRED_FROM:
            prisonNamesById.get(timelineEventResponse.contextualInfo.PRISON_TRANSFERRED_FROM) ||
            timelineEventResponse.contextualInfo.PRISON_TRANSFERRED_FROM,
        }
      : timelineEventResponse.contextualInfo,
```

I don't know if there's a more succinct way of doing it?